ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -1857,7 +1863,21 @@ - (instancetype)initWithMaximumLength:(NSInteger)maximumLength {
         _spellCheckingType = UITextSpellCheckingTypeDefault;
         _keyboardType = UIKeyboardTypeDefault;
         _multipleLines = YES;
-        _emailAddress = NO;
+    }
+    return self;
+}
+
+- (instancetype)initWithValidationExpression:(NSString *)expression invalidMessage:(NSString *)invalidMessage {
+    self = [super init];
+    if (self) {
+        _regex = expression;
+        _invalidMessage = invalidMessage;
+        _maximumLength = 0;
+        _autocapitalizationType = UITextAutocapitalizationTypeSentences;
+        _autocorrectionType = UITextAutocorrectionTypeDefault;
+        _spellCheckingType = UITextSpellCheckingTypeDefault;
+        _keyboardType = UIKeyboardTypeDefault;
+        _multipleLines = YES;
```

Comment:
The two initializers share some same settings:

```
+        _autocapitalizationType = UITextAutocapitalizationTypeSentences;
+        _autocorrectionType = UITextAutocorrectionTypeDefault;
+        _spellCheckingType = UITextSpellCheckingTypeDefault;
+        _keyboardType = UIKeyboardTypeDefault;
+        _multipleLines = YES;
```
Can you factor them out to a shared method? Maybe call it  `commonInit`?